src/localkiwixserver.cpp

```diff
@@ -41,15 +41,20 @@ LocalKiwixServer::LocalKiwixServer(QWidget *parent) :
     QVector<QString> interfaces;
     interfaces.reserve(interfacesMap.size() + 1);
     for (const auto &interfacePair : interfacesMap) {
-        QString ip = QString::fromStdString(interfacePair.second.addr);
-        interfaces.push_back(ip);
+        QString ipv4 = QString::fromStdString(interfacePair.second.addr);
+        if (!ipv4.isEmpty()) interfaces.push_back(ipv4);
+        QString ipv6 = QString::fromStdString(interfacePair.second.addr6);
+        if (!ipv6.isEmpty()) interfaces.push_back(ipv6);
     }
     std::sort(interfaces.begin(), interfaces.end());
-    interfaces.push_front(QString(gt("all")));
+    interfaces.push_front(QString(gt("ipv6")));
+    interfaces.push_front(QString(gt("ipv4")));
+    interfaces.push_front(QString(gt("all_ips")));
     for (const auto &interface : interfaces) {
         ui->IpChooser->addItem(interface);
     }
-    ui->IpChooser->setCurrentText(KiwixApp::instance()->getSettingsManager()->getKiwixServerIpAddress());
+    QString ipAddress = KiwixApp::instance()->getSettingsManager()->getKiwixServerIpAddress();
+    ui->IpChooser->setCurrentText(ipAddress == "all_ips" || ipAddress == "ipv4" || ipAddress == "ipv6" ? gt(ipAddress) : ipAddress);
```

Comment:
Please add backward compatibility by converting `"0.0.0.0"` to `"ipv4"`.

Reply:
Good catch, have to start being more aware of that

Reply:
Shouldnt it be converted to "all" though?

Reply:
I thought about that too, but chose to be conservative.

@kelson42 What's your opinion on this? Should we upgrade the old setting of the local server IP of "0.0.0.0" to all IPs (dual stack IPv4 & IPv6 mode) or just all IPv4 IPs?

Reply:
@kelson42 do you have any suggestion on how to handle this?

Reply:
Not an expert, but to me 0.0.0.0 is the same as IPv4 all... and yes woukd be better to convert,

Reply:
@kelson42 "0.0.0.0" in this case was just the internal representation. Better phrased I think the question would be "If a user of the previous implementation had chosen `All`, should that now be transferred to `All - Dual Stack Mode` or `IPv4 Only Mode`?"

Reply:
It's not an internal representation, see https://en.wikipedia.org/wiki/0.0.0.0

Reply:
Yes, that is true, but I probably did not properly phrase what I meant. I mean that in this case it is not a matter of what the setting was saved as (which is 0.0.0.0 -- because we were only working with IPv4). It is a matter of what the user was selecting which was `All`. What should now be the equivalent of the previous `All`, based on the new options?

I hope this makes more sense.

Reply:
I believe to have written already twice that it should be all IPv4. If someone wants something different it will have to change it manually.

Reply:
Clear, ok. Will proceed based on that.